apis/logging/v2beta/output_types.go

```diff
@@ -0,0 +1,316 @@
+	// '<structure-type-name>' will forward those container logs to an alternate index
+	// from that defined by the other 'structured' keys here
+	//
+	// +optional
+	EnableStructuredContainerLogs bool `json:"enableStructuredContainerLogs,omitempty"`
+}
+
+// Loki provides optional extra properties for `type: loki`
+type Loki struct {
+	// TenantKey is a meta-data key field to use as the TenantID,
+	// For example: 'TenantKey: kubernetes.namespace_name` will use the kubernetes
+	// namespace as the tenant ID.
+	//
+	// +optional
+	TenantKey string `json:"tenantKey,omitempty"`
+
+	// LabelKeys is a list of meta-data field keys to replace the default Loki labels.
+	//
+	// Loki label names must match the regular expression "[a-zA-Z_:][a-zA-Z0-9_:]*".
+	// Illegal characters in meta-data keys are replaced with "_" to form the label name.
+	// For example meta-data key "kubernetes.labels.foo" becomes Loki label "kubernetes_labels_foo".
+	//
+	// If LabelKeys is not set, the default keys are `[log_type, kubernetes.namespace_name, kubernetes.pod_name, kubernetes_host]`
+	// These keys are translated to Loki labels by replacing '.' with '_' as: `log_type`, `kubernetes_namespace_name`, `kubernetes_pod_name`, `kubernetes_host`
+	// Note that not all logs will include all of these keys: audit logs and infrastructure journal logs do not have namespace or pod name.
+	//
+	// Note: the set of labels should be small, Loki imposes limits on the size and number of labels allowed.
+	// See https://grafana.com/docs/loki/latest/configuration/#limits_config for more.
+	// You can still query based on any log record field using query filters.
+	//
+	// +optional
+	LabelKeys []string `json:"labelKeys,omitempty"`
+}
+
+// GoogleCloudLogging provides configuration for sending logs to Google Cloud Logging
+type GoogleCloudLogging struct {
+	// Only one of BillingAccountID, OrganizationID, FolderID, or ProjectID can be used to send logs.
+	// If more than one are configured, the priority is as follows
+	// 1. BillingAccountID
+	// 2. OrganizationID
+	// 3. FolderID
+	// 4. ProjectID
+	//
+	// Reference: https://cloud.google.com/billing/docs/concepts
+
+	// +optional
+	BillingAccountID string `json:"billingAccountId,omitempty"`
+
+	// +optional
+	OrganizationID string `json:"organizationId,omitempty"`
+
+	// +optional
+	FolderID string `json:"folderId,omitempty"`
+
+	// +optional
+	ProjectID string `json:"projectId,omitempty"`
+
+	//LogID is the log ID to which to publish logs. This identifies log stream.
+	LogID string `json:"logId,omitempty"`
+}
+
+// Splunk Deliver log data to Splunk’s HTTP Event Collector
+// Provides optional extra properties for `type: splunk_hec` ('splunk_hec_logs' after Vector 0.23
+type Splunk struct {
+	// Fields to be added to Splunk index. https://docs.splunk.com/Documentation/Splunk/8.0.0/Data/IFXandHEC
+	// Should be a valid JSON object
+	// +optional
+	Fields []string `json:"fields,omitempty"`
```

Comment:
I will be updating Fields description and type in the next day or so.   Here are the other fields that are missing:

	// IndexKey is a meta-data key field to use to send events to.
	// For example: 'IndexKey: kubernetes.namespace_name` will use the kubernetes
	// namespace as the index.
	// If the IndexKey is not found, the default index defined within Splunk is used.
	// Only one of IndexKey or IndexName can be defined.
	// If IndexKey && IndexName are not specified, the default index defined within Splunk is used.
	// +optional
	IndexKey string `json:"indexKey,omitempty

	// IndexName is the name of the index to send events to.
	// Only one of IndexKey or IndexName can be defined.
	// If IndexKey && IndexName are not specified, the default index defined within Splunk is used.
	// +optional
	IndexName string `json:"indexName,omitempty"

Reply:
The parts of the v2 interface that aren't changing since v1 will need to be re-synced before we ship. @jcantrill move fun with multiple API versions....